src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java

```diff
@@ -502,14 +498,45 @@ public BuildInfos getBuildInfos(UUID nodeUuid) {
         return buildInfos;
     }
 
+    private void fillInvalidateNodeInfos(NodeEntity node, InvalidateNodeInfos invalidateNodeInfos) {
+        UUID reportUuid = repositories.get(node.getType()).getReportUuid(node.getIdNode(), false);
+        if (reportUuid != null) {
+            invalidateNodeInfos.addReportUuid(reportUuid);
+        }
+
+        String variantId = repositories.get(node.getType()).getVariantId(node.getIdNode(), false);
+        if (!StringUtils.isBlank(variantId)) {
+            invalidateNodeInfos.addVariantId(variantId);
+        }
+    }
+
     @Transactional
-    public void invalidateChildrenBuildStatus(NodeEntity nodeEntity, List<UUID> changedNodes, boolean invalidateOnlyChildrenBuildStatus) {
+    public void invalidateBuild(UUID nodeUuid, boolean invalidateOnlyChildrenBuildStatus, InvalidateNodeInfos invalidateNodeInfos) {
+        List<UUID> changedNodes = new ArrayList<>();
+        UUID studyId = getStudyUuidForNodeId(nodeUuid);
+
+        nodesRepository.findById(nodeUuid).ifPresent(n -> {
+            if (!invalidateOnlyChildrenBuildStatus) {
+                repositories.get(n.getType()).invalidateBuildStatus(nodeUuid, changedNodes);
```

Comment:
Why we do not use updateBuildStatus ?

Reply:
Using invalidateBuildStatus avoids a db writing in case current build status is different from BUILT